h2/src/main/org/h2/table/Table.java

```diff
@@ -620,7 +620,7 @@ public void dropMultipleColumnsConstraintsAndIndexes(Session session,
     }
 
     public Row getTemplateRow() {
-        return database.createRow(new Value[columns.length], Row.MEMORY_CALCULATE);
+        return database.createRow(new Value[getColumns().length], Row.MEMORY_CALCULATE);
```

Comment:
I have introduced the AbstractTable class, but that required quite a lot of changes and it is a bit strange to have a AbstractTable class as well as a abstract Table class, but I couldn't think of a better name either.